### PR TITLE
Update system-exploration directory

### PR DESCRIPTION
**Pull request by SorinAlexB**
# Prerequisite Checklist

- [ ] Read the [contribution guidelines](https://github.com/open-education-hub/ccas-internal/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [ ] Tested your changes against relevant architectures and platforms;
- [ ] Updated relevant documentation (if needed).

## Description of changes

Transitioned the 'system-exploration' directory to meet the requirements of the OpenEdu Methodology.

### CODE REVIEW

File: chapters/scratch-linux/system-exploration/drills/tasks/doge/README.md
Comment (teodutu) [inline chapters/scratch-linux/system-exploration/drills/tasks/doge/README.md:1]: Move this content to `guides/`

File: chapters/scratch-linux/system-exploration/reading/README.md
Comment (teodutu) [inline chapters/scratch-linux/system-exploration/reading/README.md:1]: Split this file into:
- `introduction.md` (with the introduction, reminders and prerequisites)
- `exploration-tools.md`: https://security-summer-school.github.io/essentials-security/Scratch%20Linux/System%20Exploration/#continuing-our-exploration
- `file-inspection.md`: https://security-summer-school.github.io/essentials-security/Scratch%20Linux/System%20Exploration/#inspecting-files
- `summary.md`: https://security-summer-school.github.io/essentials-security/Scratch%20Linux/System%20Exploration/#summary
- `further-reading.md`: https://security-summer-school.github.io/essentials-security/Scratch%20Linux/System%20Exploration/#further-reading

**Review by teodutu — CHANGES_REQUESTED** [review]
Fix the super-linter errors [1]. I know some of them are annoying (especially those related to shell scripts), but most do make sense. In addition, reword your commit to `scratch-linux/system-exploration: Rework session content according to OpenEdu methodology` then add a description of your changes below.

[1] https://github.com/security-summer-school/essentials-security/actions/runs/9626390655/job/26552309089?pr=9

**Review by teodutu — APPROVED** [review]
:ok_hand:

**Comment by teodutu** [comment]
@Ryx076 add your review so we can merge this after your feedback is implemented.

File: chapters/scratch-linux/system-exploration/drills/tasks/not-your-doge/solution/solution.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+#! /usr/bin/python
+
+file = open("../public/not-doge.pnm", "rb")
```
Comment (Ryx076) [inline chapters/scratch-linux/system-exploration/drills/tasks/not-your-doge/solution/solution.py]:
```suggestion
file = open("../../support/not-doge.pnm", "rb")
```

File: chapters/scratch-linux/system-exploration/drills/tasks/surgical-precision/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Surgical Precision
+
+There are many false flags out there.
+Answer the questions and find the **real** flags.
+The quizzes cover subjects discussed today and during the previous session.
+Think of them as a recap.
+
+The answer to each of the riddles in the files `question-*` from the `activities/surgical-precision/public` is the name of one of the given files.
```
Comment (Ryx076) [inline chapters/scratch-linux/system-exploration/drills/tasks/surgical-precision/README.md]:
```suggestion
The answer to each of the riddles in the files `question-*` from the `drills/tasks/surgical-precision/support` is the name of one of the given files.
```

File: chapters/scratch-linux/system-exploration/guides/doge/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Doge
+
+The best way to showcase the `strings` command is to use it in order to find our first flag for today.
+Head to the `tasks/doge/support` folder and take a look at the image you've been given.
```
Comment (Ryx076) [inline chapters/scratch-linux/system-exploration/guides/doge/README.md]:
```suggestion
Head to the `guides/doge/support` folder and take a look at the image you've been given.
```

File: chapters/scratch-linux/system-exploration/guides/doge/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Since this section is dedicated to the `strings` command, we'll run this command on our `doge.jpg` file:
+
+```bash
+root@kali:~/essentials/system-exploration/activities/doge/public# strings doge.jpg
```
Comment (Ryx076) [inline chapters/scratch-linux/system-exploration/guides/doge/README.md]:
```suggestion
root@kali:~/essentials-security/chapters/scratch-linux/system-exploration/guides/doge/public# strings doge.jpg
```

Comment (Ryx076) [reply chapters/scratch-linux/system-exploration/guides/doge/README.md]: and the rest

File: chapters/scratch-linux/system-exploration/reading/file-inspection.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+First, we want to get some more detailed information about what type of binary file we are dealing with specifically.
+`ls` already gives us information such as the file's name, size and permissions.
+This is all fine, but this information is common to all files.
+Whether we're dealing with an image, or with an executable file `ls` won't tell us.
+
+But `file` does.
+`file` works by reading a file's header (the first few bytes at the beginning of the file, which hold information about its format and type).
+Thus, it is capable of outputting more precise information than `ls`.
+Let's test it using one of today's challenges, `activities/05-challenge_not-your-doge/public/not-doge.pnm`.
```
Comment (Ryx076) [inline chapters/scratch-linux/system-exploration/reading/file-inspection.md]:
```suggestion
Let's test it using one of today's challenges, `drills/tasks/not-your-doge/support/not-doge.pnm`.
```

File: chapters/scratch-linux/system-exploration/reading/file-inspection.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+```bash
+root@kali:~/essentials/system-exploration# file activities/05-challenge_not-your-doge/public/not-doge.pnm
```
Comment (Ryx076) [inline chapters/scratch-linux/system-exploration/reading/file-inspection.md]:
```suggestion
root@kali:~/essentials-security/chapters/scratch-linux/system-exploration# file drills/tasks/not-your-doge/support/not-doge.pnm
```

File: chapters/scratch-linux/system-exploration/reading/file-inspection.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+activities/05-challenge_not-your-doge/public/not-doge.pnm: Netpbm image data, size = 500 x 590, rawbits, pixmap
```
Comment (Ryx076) [inline chapters/scratch-linux/system-exploration/reading/file-inspection.md]:
```suggestion
drills/tasks/not-your-doge/support/not-doge.pnm: Netpbm image data, size = 500 x 590, rawbits, pixmap
```

**Review by Ryx076 — CHANGES_REQUESTED** [review]
Move `drills/tasks/surgical-precision/public` to `drills/tasks/surgical-precision/support`

**Review by Ryx076 — APPROVED** [review]

**Pull request merged**